pkg/asset/machines/baremetal/machines.go

```diff
@@ -61,6 +60,5 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 }
 
 func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, userDataSecret string) *libvirtprovider.LibvirtMachineProviderConfig {
-	// FIXME: baremetal
-	return &libvirtprovider.LibvirtMachineProviderConfig{}
+	return &baremetalprovider.BareMetalMachineProviderSpec{}
```

Comment:
My guess is we should populate the fields in BareMetalMachineProviderSpec, like other providers do: https://github.com/openshift-metal3/kni-installer/blob/master/pkg/asset/machines/aws/machines.go#L78-L111, but I'm not 100% sure.

Reply:
Yeah, though we don't necessarily have to block on that for the first PR.  The current cluster-api-provider-baremetal we're running will ignore it either way.

We need to populate some fields here before we can rebase to 4.2 though, as that will include the version that expects some data in here